security/src/main/java/org/kaazing/gateway/security/auth/FileLoginModule.java

```diff
@@ -234,19 +226,19 @@ private void cleanState() {
         password = null;
     }
 
-    private boolean attemptAuthenticate(boolean useSharedState)
+    private void attemptAuthenticate(boolean useSharedState)
         throws LoginException {
 
         getUsernamePassword(useSharedState);
 
         if (username == null) {
-            LOG.debug("[FileLoginModule] " + "username not found, returning false");
-            return false;
+            LOG.debug("[FileLoginModule] username not found");
+            throw new LoginException("Username not found");
```

Comment:
It seems to me that in the Http Security Filter, we should log (at debug) the details of the Login Exception.  If that is the case, I don't know if the Log's here are necessary... (thoughts?, if you agree there are probably several spots we could change this).    But we would ideally also include the exception information (if any) from the call getUsernamePassword in this LoginException (I would think).